app/kubemci/pkg/kubeutils/utils.go

```diff
@@ -93,3 +103,120 @@ var getClientset = func(kubeconfigPath, contextName string) (kubeclient.Interfac
 
 	return kubeclient.NewForConfigOrDie(clientConfig), nil
 }
+
+type KubernetesInformer struct {
+	ctx       *ingresscx.ControllerContext
+	svcSynced cache.InformerSynced
+	podSynced cache.InformerSynced
+	svcLister ingressct.StoreToServiceLister
+	podLister ingressct.StoreToPodLister
+}
+
+func NewKubernetesInformer(context *ingresscx.ControllerContext) KubernetesInformer {
+	b := KubernetesInformer{}
+	b.ctx = context
+	b.svcSynced = context.ServiceInformer.HasSynced
+	b.podSynced = context.PodInformer.HasSynced
+	b.svcLister.Indexer = context.ServiceInformer.GetIndexer()
+	b.podLister.Indexer = context.PodInformer.GetIndexer()
+	return b
+}
+
+func (b *KubernetesInformer) Init() {
+	b.ctx.Start()
+	stopCh := make(chan struct{})
+	if !cache.WaitForCacheSync(stopCh, b.svcSynced, b.podSynced) {
+		fmt.Println("Unable to sync kubernetes informers")
+	}
+
+	fmt.Println("Synced", b.svcSynced(), b.podSynced())
+}
+
+// GetProbe returns a probe that's used for the given nodeport
+func (t *KubernetesInformer) GetProbe(port ingressbe.ServicePort) (*api_v1.Probe, error) {
+	sl := t.svcLister.List()
+
+	// Find the label and target port of the one service with the given nodePort
+	var service api_v1.Service
+	var svcPort api_v1.ServicePort
+	var found bool
+OuterLoop:
+	for _, as := range sl {
+		service = *as.(*api_v1.Service)
+		for _, sp := range service.Spec.Ports {
+			svcPort = sp
+			// only one Service can match this nodePort, try and look up
+			// the readiness probe of the pods behind it
+			if int32(port.Port) == sp.NodePort {
```

Comment:
We have already done this part while computing ServicePort: https://github.com/GoogleCloudPlatform/k8s-multicluster-ingress/blob/ecff59d1b6493bc28bab02cb4de74c23e9b047b6/app/kubemci/pkg/gcp/loadbalancer/loadbalancersyncer.go#L365
Here we can directly fetch service with name port.SvcName to get the required service.

Reply:
Ah cool, thanks!